### PR TITLE
feat(gh-actions): add ios build workflow

### PR DESCRIPTION
**Pull request by ferdicus**
## Description
After adding the detox steps in an earlier PR.
Why not add a build step for iOS.

This will (on every push) build the iOS `example` build and run our (short and small) detox tests on it.

~I'm expecting, that this won't run without ~additional~ extensive tweaks 😅~

Here goes nothing 🤞🏿 

### Additional info
* additional e2e test to come
* removal of unit tests from `/example` 
  (unit test to be in the root project)

### CODE REVIEW

File: example/package.json [hunk 1]
```diff
@@ -12,9 +12,10 @@
     "lint": "eslint ./src",
     "postinstall": "node ./scripts/set_access_token.js && jetifier",
     "purge:android": "(cd android && ./gradlew --stop) && rm -rf ~/.gradle/caches/ android/app/build",
-    "purge:ios": "rm -rf ios/Pods/* ~/Library/Caches/CocoaPods Pods ~/Library/Developer/Xcode/DerivedData && pod cache clean --all",
+    "purge:ios": "rm -rf ios/Pods/* ios/build ~/Library/Caches/CocoaPods ~/Library/Developer/Xcode/DerivedData && pod cache clean --all",
     "purge:js": "rm -rf node_modules && yarn cache clean && watchman watch-del-all",
-    "purge": "yarn purge:js && yarn purge:android && yarn purge:ios"
+    "purge": "yarn purge:js && yarn purge:android && yarn purge:ios",
+    "e2e:ios": "yarn start & METRO_BUNDLER_PID=$!; yarn detox test --debug-synchronization 200; DETOX_EXIT_CODE=$?; kill $METRO_BUNDLER_PID; exit $DETOX_EXIT_CODE"
```
Comment (mattijsf) [inline example/package.json:18]: `yarn start` should not be needed. Metro will be started automatically via the "Start Packager" build phase on the Example project. I'm also not familiar with the `METRO_BUNDLER_PID` `DETOX_EXIT_CODE` variables and related commands. `yarn detox test` should be sufficient to bootstrap the whole process. See also: https://github.com/wix/Detox/blob/master/docs/Guide.RunningOnCI.md#step-2-add-build-and-test-commands-to-your-ci-script

Edit:
I noticed this is probably there because we're testing against a Debug build in Detox. Any reason for not testing against a Release build?

**Review by mattijsf — APPROVED** [review]

File: example/package.json (hunk 1 above)
Comment (ferdicus) [reply example/package.json]: hey, thanks for having a look 🙇🏿 

I've initially tried building a `Release` build, however it failed with this message:
`building for iOS Simulator, but linking in object file built for iOS, for architecture arm64`, which I now fixed in [this commit](https://github.com/react-native-mapbox-gl/maps/pull/1271/commits/a6de14cd662a4bb7c6534c0af384079fd5593420)  - so `Release` build it is now 👍🏿 

I also found, that the builds take waaaay too long, thus I removed the `Flipper` integration by commenting it our in the [Podfile](https://github.com/react-native-mapbox-gl/maps/pull/1271/commits/675a7160d648cd4ada1bbb3a9827941e3ccf8c03) - if someone needs it for testing purposed, just comment it back in.

Lastly, `METRO_BUNDLER_PID` is just a variable for `$!` which is [`the PID of the most recent background command.`](https://unix.stackexchange.com/a/287445) and `DETOX_EXIT_CODE` is just a variable for `$?` which is [`the return value of the last executed command`](is used to find the return value of the last executed command)


Long story short:
* works with `Release`
* takes less due to dumping `Flipper`
* does not need convoluted test script

Thanks
